### PR TITLE
[bugfix] Ensure that job options passed from the command-line are always the last in the generated job script

### LINKED ISSUE #1700: `self.job.options` override any job options passed from the command-line
Currently, if I have a test that does:

```python
@rfm.run_before('run')
def set_job_options(self):
    self.job.options = ['--foo']
```

And then run reframe with `-J bar=X`, the `--bar=X` will never be passed to the job script.

### PR DESCRIPTION
**Pull request by rsarm**
Closes #1700

### CODE REVIEW

**Review by teojgo — APPROVED** [review]
lgtm

File: reframe/core/schedulers/__init__.py [hunk 1]
```diff
@@ -180,7 +180,8 @@ def __init__(self,
         self.num_cpus_per_task = None
         self.use_smt = None
         self.time_limit = None
-        self.options = list(sched_options) if sched_options else []
+        self.sched_options = list(sched_options) if sched_options else []
```
Comment (vkarak) [inline reframe/core/schedulers/__init__.py:183]: `sched_options` is a bit confusing as a name here. What about `cli_options`?

**Review by vkarak — CHANGES_REQUESTED** [review]
Lgtm except that we should probably use a better name than `sched_options` inside the `Job`.

**Review by vkarak — APPROVED** [review]
lgtm

**Pull request merged**